### PR TITLE
Survival lighter.sc - It works like the previous, but in a 128 box around player

### PR DESCRIPTION
**Pull request by Ghoulboy78**
and it consumes torches in survival and gives weird error messages if you run out of torches, so yay
and you can add rand if you want, but i found it kinda pointless, the user can chooze

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
however, it looks less cool than the previous, ill say that much

File: programs/survival/survival_lighter.sc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+__on_player_uses_item(player, item_tuple, hand)->(
+	l(item,count,nbt)=item_tuple;
+	if(hand=='mainhand'&&item == 'torch'&&nbt==null,
+		print('Started placing torches ... ');//You can turn off feedback
+		l(cx,cy,cz)=pos(player);
+		scan(cx,cy,cz,128,128,128,
+			if(!inventory_find(player,'torch'),return(print('You ran out of torches')));//it gives weird error messages in surival for some reason, but works otherwise.
+			lpos=pos(_);
+			if((block(lpos)=='air'||block(lpos)=='cave_air')&&light(lpos)<8&&solid(pos_offset(lpos, 'down')),//&&!rand(10), optional
```
Comment (gnembon) [inline programs/survival/survival_lighter.sc]: air(pos) will check for you 'air', 'cave_air' as well as 'void_air' in one expression

Comment (Ghoulboy78) [reply programs/survival/survival_lighter.sc]: done

**Comment by gnembon** [comment]
Why don't you just add survival check for item usage to the previous one? - its a cool demonstration no how to control falling block entities in scarpet plus it looks cool - the only thing its missing is checks for survival to remove the item (making sure the torch stack in the hand is the last to be used).

**Comment by Ghoulboy78** [comment]
why should the item in the mainhand be the last to be used? and ok, perhaps i will try und find a way

**Comment by Ghoulboy78** [comment]
you shouldn't have man, I could have done it! I mean, I just had a little extra work this week.

**Pull request closed without merging**